src/cascadia/TerminalSettingsEditor/Appearances.cpp

```diff
@@ -49,6 +49,88 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         }
     }
 
+    double AppearanceViewModel::LineHeight() const noexcept
+    {
+        const auto cellSizeY = _appearance.SourceProfile().CellSizeY();
+        const auto str = cellSizeY.c_str();
+
+        auto& errnoRef = errno; // Nonzero cost, pay it once.
+        errnoRef = 0;
+
+        wchar_t* end;
+        const auto value = std::wcstod(str, &end);
+
+        return str == end || errnoRef == ERANGE ? NAN : value;
+    }
+
+    void AppearanceViewModel::LineHeight(const double value)
+    {
+        wchar_t buffer[16];
+        std::wstring_view view;
+
+        if (value >= 0.1 && value <= 10.0)
+        {
+            auto length = gsl::narrow<size_t>(swprintf_s(&buffer[0], std::size(buffer), L"%.6f", value));
+            const auto separator = std::wstring_view{ &buffer[0], length }.find(L'.');
+
+            if (separator != std::wstring_view::npos)
+            {
+                for (; length > separator && buffer[length - 1] == L'0'; --length)
+                {
+                }
+                // winrt::hstring expects a null-terminated string
+                buffer[length] = L'\0';
+            }
+
+            view = { &buffer[0], length };
+        }
+
+        const auto profile = _appearance.SourceProfile();
+
+        if (profile.CellSizeY() != view)
+        {
+            if (view.empty())
+            {
+                profile.ClearCellSizeY();
+            }
```

Comment:
When is this hit?
Don't we still want to set it? Doesn't `ClearCellSizeY()` make it so that we're inheriting? Whereas here, the user explicitly put a value in, so we don't want to inherit?

Reply:
When the value is outside of the valid range. For instance when user clears the field and the `NumberBox` calls us then it'll pass `NAN`. `NAN` is not a valid line height so I'd either have to store an empty string or `null` in JSON. I felt like both would be weird so now I just delete the override from the settings when the user clears the `NumberBox` and hits save.